### PR TITLE
AST-9929: added project id and team filter

### PR DESCRIPTION
**Pull request by serhii-stanislavskyi**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../CODE-OF-CONDUCT.md). Please see the [contributing guidelines](../CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

**Review by ecasanova-checkmarx — APPROVED** [review]

File: internal/integration/rest/utils.go [hunk 1]
```diff
@@ -1,38 +1,83 @@
...
-			if project.ID == projectOData.ID {
-				addDateAndCustomFields(project, projectOData)
-				break
-			}
-		}
-	}
-
-	return projects
+// GetFilterForProjects get filter string for projects list
+func GetFilterForProjects(fromDate, teamName, projectIds string) string {
```
Comment (constantinoantunes1) [inline internal/integration/rest/utils.go:20]: Please add unit tests.
Looks like we need at least:
 - empty team + empty projectids
 - team + empty projectids
 - empty team + project ids
 - team + project ids

File: internal/integration/rest/utils.go [hunk 2]
```diff
@@ -1,38 +1,83 @@
...
+	}
+	if projectIds == "" {
+		return fmt.Sprintf("CreatedDate gt %s and %s", fromDate, getTeamFilter(teamName))
+	}
+	return fmt.Sprintf("CreatedDate gt %s and %s and %s", fromDate, getTeamFilter(teamName),
+		getProjectIdsFilter(projectIds))
+}
+
+// GetFilterForProjectsWithLastScan get filter string for projects list with last scan
+func GetFilterForProjectsWithLastScan(fromDate, teamName, projectIds string) string {
```
Comment (constantinoantunes1) [inline internal/integration/rest/utils.go:35]: Please add unit tests to this and other functions in this file

File: internal/integration/rest/utils_test.go [hunk 3]
```diff
@@ -7,42 +7,88 @@ import (
 )
 
 func TestExtendProjects(t *testing.T) {
```
Comment (constantinoantunes1) [inline internal/integration/rest/utils_test.go:9]: ExtendProjects doesn't seem to exist anymore.

File: internal/process_test.go [hunk 4]
```diff
@@ -1250,25 +1278,29 @@ func TestFetchProjects(t *testing.T) {
 			}}}
 		exporter := mock_app_export.NewMockExporter(gomock.NewController(t))
 		client := mock_integration_rest.NewMockClient(gomock.NewController(t))
-		client.EXPECT().GetProjects().Return(projects, nil).Times(1)
+		client.EXPECT().GetProjects(gomock.Any(), gomock.Eq(teamName), gomock.Eq(projectsIds), gomock.Eq(0),
```
Comment (constantinoantunes1) [inline internal/process_test.go:1281]: No need for `gomock.Eq(x)` you can just use `x` directly.

File: internal/process_test.go [hunk 5]
```diff
@@ -1242,6 +1268,8 @@ func TestExportResultsToFile(t *testing.T) {
 }
 
 func TestFetchProjects(t *testing.T) {
```
Comment (constantinoantunes1) [inline internal/process_test.go:1270]: Could you please add tests for the scenario of many projects, over many pages, that we filter using teamName and projectIds?

**Review by constantinoantunes1 — CHANGES_REQUESTED** [review]

File: internal/integration/rest/utils.go (hunk 1 above)
Comment (serhii-stanislavskyi) [reply internal/integration/rest/utils.go:20]: I covered such test cases

File: internal/integration/rest/utils_test.go (hunk 3 above)
Comment (serhii-stanislavskyi) [reply internal/integration/rest/utils_test.go:9]: I'll fix this :)

File: internal/process_test.go (hunk 4 above)
Comment (serhii-stanislavskyi) [reply internal/process_test.go:1281]: Ok

File: internal/process_test.go (hunk 5 above)
Comment (serhii-stanislavskyi) [reply internal/process_test.go:1270]: Ok

File: internal/integration/rest/utils.go (hunk 2 above)
Comment (serhii-stanislavskyi) [reply internal/integration/rest/utils.go:35]: I've already added these test cases

File: internal/process_test.go (hunk 5 above)
Comment (serhii-stanislavskyi) [reply internal/process_test.go:1270]: About teamName and projectIds filter there are no sense to make unit tests here because filter is mocked here

**Review by constantinoantunes1 — APPROVED** [review]

**Pull request merged**